### PR TITLE
cli: Use clikt instead of JCommander for command line parsing

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

File: cli/src/funTest/kotlin/OrtMainTest.kt [hunk 1]
```diff
@@ -56,54 +56,51 @@ class OrtMainTest : StringSpec() {
...
                 "-i", inputDir.path,
                 "-o", File(outputDir, "gradle").path
             )
-            val iterator = runResult.stdout.iterator()
+            val iterator = stdout.iterator()
             while (iterator.hasNext()) {
                 if (iterator.next() == "The following package managers are activated:") break
             }
 
-            runResult.exitCode shouldBe 0
```
Comment (mnonnenmacher) [inline cli/src/funTest/kotlin/OrtMainTest.kt]: Why do you not test the exit codes anymore?

File: cli/src/main/kotlin/commands/DownloaderCommand.kt [hunk 2]
```diff
@@ -114,7 +114,7 @@ object DownloaderCommand : CommandWithHelp() {
     private var archive = false
 
     @Parameter(
-        description = "The type of data entities from the ORT file's analyzer result to download.",
+        description = "The data entities from the ORT file's analyzer result to limit downloads to.",
```
[the remark this answers is not in the document]
Comment (sschuberth) [reply cli/src/main/kotlin/commands/DownloaderCommand.kt]: Done.

File: cli/src/main/kotlin/OrtMain.kt [hunk 3]
```diff
@@ -51,128 +58,86 @@ const val TOOL_NAME = "ort"
...
+    private inner class OrtHelpFormatter : CliktHelpFormatter(requiredOptionMarker = "*", showDefaultValues = true) {
+        override fun formatHelp(
+            prolog: String,
+            epilog: String,
+            parameters: List<HelpFormatter.ParameterHelp>,
+            programName: String
+        ) =
+            buildString {
+                // If help is invoked without a subcommand, the main run() is not invoked and no header is printed, so
+                // we need to to that manually here.
```
[the remark this answers is not in the document]
Comment (sschuberth) [reply cli/src/main/kotlin/OrtMain.kt]: Done.

File: cli/src/funTest/kotlin/OrtMainTest.kt (hunk 1 above)
Comment (sschuberth) [reply cli/src/funTest/kotlin/OrtMainTest.kt]: Because `clikt`'s `parse()` throws `UsageError` instead.

**Pull request merged**